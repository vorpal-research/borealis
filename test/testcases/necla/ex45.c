#include <stdlib.h>
#include <string.h>

// parse the string based on the token char c.
char **parse_token(char *str, char c)
{
   int len = strlen(str);
   int loc = 0;
   int num = 0;
   int i;
   char **res;
   
   for (i = 0; i < len; i++)
      if (str[i] == c)
            num++;

   res = (char **) malloc(num * sizeof (char*));
   for (i = 0; i < len; i++)
   {
      if (str[i] == c) {
        char *temp_str = (char *) calloc (i-loc+1, 1); // the last char is 0
        memcpy(temp_str, str+loc, i-loc);
        res[num++] = temp_str;
        loc = i+1;
      }
             
   }

   return res;
   
}

int main(char *str, char c){

   char **tokens;
  
   ASSUME(strlen(str) > 0);
   
   tokens = parse_token(str, c);
   
   return 1;
}
