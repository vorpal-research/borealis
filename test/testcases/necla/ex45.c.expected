
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex45.c",
			"loc" : 
			{
				"col" : 7,
				"line" : 14
			}
		},
		"type" : "BUF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex45.c",
			"loc" : 
			{
				"col" : 7,
				"line" : 21
			}
		},
		"type" : "BUF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex45.c",
			"loc" : 
			{
				"col" : 10,
				"line" : 24
			}
		},
		"type" : "BUF-01"
	}
]
